Mainnet/Identity/IdentityContracts/IdentityProvider.cs

```diff
@@ -0,0 +1,143 @@
+﻿using Stratis.SmartContracts;
+
+/// <summary>
+/// A constrained version of ERC780.
+/// This contract only holds claims from one identity provider, the "Owner".
+/// </summary>
+public class IdentityProvider : SmartContract
+{
+    private Address Owner
+    {
+        get
+        {
+            return this.PersistentState.GetAddress(nameof(Owner));
+        }
+        set
+        {
+            this.PersistentState.SetAddress(nameof(Owner), value);
+        }
+    }
+
+    public IdentityProvider(ISmartContractState state) : base(state)
+    {
+        this.Owner = Message.Sender;
+    }
+
+    public void ChangeOwner(Address newOwner)
+    {
+        Assert(this.Owner == Message.Sender);
+        this.Owner = newOwner;
+    }
+
+    public void AddClaim(Address issuedTo, uint topic, byte[] data)
+    {
+        Assert(this.Owner == Message.Sender);
+
+        byte[] oldData = this.GetClaim(issuedTo, topic);
+        bool exists = oldData != null;
+
+        // This exact claim value is stored already. Nothing to change.
+        if (exists && ByteArrayCompare(oldData, data))
```

Comment:
The byte array comparison allows us to know if the claim has been changed. Is that valuable, or could we just log `ClaimChanged` any time the method is invoked, even if the claim is the same?

The downsides I see are added complexity by requiring the byte array comparison, which is unbounded and could potentially consume `GasLimitMaximum` if the arrays are big enough.

Reply:
We can leave this responsibility to client and save some gas so client can compare it with previous data if it really wants to know it is new data or not.

Reply:
Up to you to decide. I would push this to the client myself and log `OldData` so the client can perform the comparison :)